TESTS/host_tests/timing_drift_auto.py

```diff
@@ -16,91 +16,103 @@
-    ticks = []
-    start_time = None
-    finish_time = None
-    dut_seconds_passed = None
-    total_time = None
-    total_drift = None
-    average_drift = None
-    
+    mega = 1000000.0
+    max_measurement_time = 180
+
+    # this value is obtained for measurements when there is 0 transport delay and we want accurancy of 5%
+    time_measurement_multiplier = 80
+
     def _callback_result(self, key, value, timestamp):
         # We should not see result data in this test
         self.__result = False
 
     def _callback_end(self, key, value, timestamp):
         """ {{end;%s}}} """
         self.log("Received end event, timestamp: %f" % timestamp)
-        self.notify_complete(result=self.result(print_stats=True))
+        self.notify_complete(result=self.result())
 
+    def _callback_timing_drift_check_start(self, key, value, timestamp):
+        self.round_trip_base_start = timestamp
+        self.send_kv("base_time", 0)
 
-    def _callback_tick(self, key, value, timestamp):
-        """ {{tick;%d}}} """
-        self.log("tick! %f" % timestamp)
-        self.ticks.append((key, value, timestamp))
+    def _callback_base_time(self, key, value, timestamp):
+        self.round_trip_base_end = timestamp
+        self.device_time_base = float(value)
+        self.round_trip_base_time = self.round_trip_base_end - self.round_trip_base_start
 
+        self.log("Device base time {}".format(value))
+        measurement_stretch = (self.round_trip_base_time * self.time_measurement_multiplier) + 5
+
+        self.log("Time required to measure drift accurately  : {} ".format(measurement_stretch))
+        if measurement_stretch > self.max_measurement_time:
+            self.log("Measurement time is high to accurately qualify device timer")
+        else:
+            time.sleep(measurement_stretch)
+        self.round_trip_final_start = time.time()
+        self.send_kv("final_time", 0)
+
+    def _callback_final_time(self, key, value, timestamp):
+        self.round_trip_final_end = timestamp
+        self.device_time_final = float(value)
+        self.round_trip_final_time = self.round_trip_final_end - self.round_trip_final_start
+        self.log("Device final time {} ".format(value))
+
+        # compute the test results and send to device
+        results = "pass" if self.compute_parameter() else "fail"
+        self.send_kv(results, "0")
 
     def setup(self):
-        self.register_callback("end", self._callback_end)
-        self.register_callback('tick', self._callback_tick)
+        self.register_callback('timing_drift_check_start', self._callback_timing_drift_check_start)
+        self.register_callback('base_time', self._callback_base_time)
+        self.register_callback('final_time', self._callback_final_time)
 
+    def compute_parameter(self, failure_criteria=0.05):
+        nominal_time = (self.round_trip_final_start + self.round_trip_final_end) / 2.0 - (self.round_trip_base_start + self.round_trip_base_end) / 2.0
```

Comment:
Can you add any comments to this host files, there are lto of magic numbers, computations. It would explain what it does, for future would help to edit (at least me :-) ).